hdeps/requirements.py

```diff
@@ -2,24 +2,36 @@
 # thing here to avoid extra deps or fragile APIs, at the expense of missing some
 # deps and false-positives.
 
+import logging
+
 from glob import glob
 from pathlib import Path
 from typing import Iterator
 
 from packaging.requirements import Requirement
 from packaging.utils import canonicalize_name
 
+LOG = logging.getLogger(__name__)
+
 # These all have iter- prefixes because I expect a more public api to pick a
 # couple and return sets instead.
 
+SHOWN_IGNORE_MESSAGE = False
+
 
 def _iter_simple_requirements(path: Path) -> Iterator[Requirement]:
+    global SHOWN_IGNORE_MESSAGE
     for line in path.read_text().splitlines():
         line = line.split("#", 1)[0].strip()
         if not line:
             continue
         if line.startswith("-"):
-            print("Ignoring", line)
+            if not SHOWN_IGNORE_MESSAGE:
+                LOG.warning(
+                    "Non-simple requirements are ignored (this message only prints once)"
```

Comment:
I think this doesn't need to be a warning, it could simply be a DEBUG level message.
Warning implies user can do something to fix it, but I don't think they can or will.

They could clean up stuff like `--index-url`, but a `-r` within a reqs file, I don't think they ever will (and that shouldn't bother them with a WARNING)

Reply:
There are a bunch of constructs where `hdeps` not parsing/understanding it affects what would be found as deps though, including `-e`, `-c`, `-r`, `--pre`.  I feel like that's worth notifying the user.  Is there another way you'd suggest?

Reply:
Good point, that's more like a HEADS_UP then than a WARNING.
I don't have a better suggestion... would involve actually supporting those I guess, because you're right, they could affect the results